applications/FluidDynamicsApplication/custom_conditions/navier_stokes_wall_condition.cpp

```diff
@@ -81,6 +81,12 @@ void NavierStokesWallCondition<TDim,TNumNodes>::CalculateLocalSystem(MatrixType&
     if (rRightHandSideVector.size() != MatrixSize)
         rRightHandSideVector.resize(MatrixSize, false); //false says not to preserve existing storage!!
 
+    // Check that parents have been computed
+    // These are required to retrieve the material properties and the viscous stress
+    auto& parentElement = this->GetValue(NEIGHBOUR_ELEMENTS);
+    KRATOS_ERROR_IF(parentElement.size() > 1) << "A condition was assigned more than one parent element." << std::endl;
+    KRATOS_ERROR_IF(parentElement.size() == 0) << "A condition was NOT assigned a parent element. Please execute the check_and_prepare_model_process_fluid process." << std::endl;
+
```

Comment:
this seems expensive
why not doing it in `Check` or at least only in debug?

Reply:
The point is that right now the neighbors are not always computed by default. After having discussed about this, which in my experience is a potential source of memory errors, we decided to move to the safe solution that is always compute the condition neighbors by default (note that this is done once).

Having said this, I'd keep the current check until we change de default behavior. Then we can change it to be checked only in debug mode.